OSD: Configs without ESC sensing fails to build

```
obj/main/ANYTARGET/io/osd.o: In function `osdShowStats':
${TOPDIR}/./src/main/io/osd.c:1560: undefined reference to `osdConvertTemperatureToSelectedUnit'
${TOPDIR}/./src/main/io/osd.c:1560: undefined reference to `osdGetTemperatureSymbolForSelectedUnit'
collect2: error: ld returned 1 exit status
make[1]: *** [obj/main/betaflight_ANYTARGET.elf] Error 1
make: *** [hex] Error 2
$
```

https://github.com/betaflight/betaflight/blob/70723bdcfe1d6a0fcde3df73d44dce8d8edc0240/src/main/io/osd.c#L1554-L1567

These lines should be conditionalized with `1559 #if defined(USE_ESC_SENSOR) && defined(USE_ESC_SENSOR_INFO)` ? Both? Either?

--- a/src/main/io/osd.c
+++ b/src/main/io/osd.c
@@ -1556,6 +1556,7 @@ static void osdShowStats(uint16_t endBatteryVoltage)
         osdDisplayStatisticLabel(top++, "MAX G-FORCE", buff);
     }
 
+#ifdef USE_ESC_SENSOR
     if (osdStatGetState(OSD_STAT_MAX_ESC_TEMP)) {
         tfp_sprintf(buff, "%3d%c", osdConvertTemperatureToSelectedUnit(stats.max_esc_temp * 10) / 10, osdGetTemperatureSymbolForSelectedUnit());
         osdDisplayStatisticLabel(top++, "MAX ESC TEMP", buff);
@@ -1565,6 +1566,7 @@ static void osdShowStats(uint16_t endBatteryVoltage)
         itoa(stats.max_esc_rpm, buff, 10);
         osdDisplayStatisticLabel(top++, "MAX ESC RPM", buff);
     }
+#endif
 
 }
 

